src/app/addons/paymentwall/func.php

```diff
@@ -27,50 +27,73 @@ function fn_paymentwall_initPaymentwallSdk($projectKey, $secretKey, $apiType = P
 {
-    $defaultAttr = array(
-        'width' => '100%',
-        'height' => 400,
-        'frameborder' => 0
-    );
-
     fn_paymentwall_initPaymentwallSdk($paymentInfo['key'], $paymentInfo['secret']);
 
-    $realPrice = fn_paymentwall_getRealPrice($orderInfo);
-
     $widget = new Paymentwall_Widget(
         empty($orderInfo['user_id']) ? $orderInfo['ip_address'] : $orderInfo['user_id'],
         $paymentInfo['widget_type'],
         array(
             new Paymentwall_Product(
                 $orderInfo['order_id'],
-                $realPrice,
+                fn_paymentwall_getRealPrice($orderInfo),
                 $orderInfo['secondary_currency'],
                 'Order #' . $orderInfo['order_id']
             )
-        ), array(
-        'email' => $orderInfo['email'],
-        'payment_id' => $orderInfo['payment_id'],
-        'integration_module' => 'cs_cart',
-        'test_mode' => $paymentInfo['test_mode']
-    ));
+        ),
+        array_merge(
+            array(
+                'email' => $orderInfo['email'],
+                'payment_id' => $orderInfo['payment_id'],
+                'integration_module' => 'cs_cart',
+                'test_mode' => $paymentInfo['test_mode']
+            ),
+            fn_paymentwall_get_shipping_information($orderInfo)
+        ));
 
     // Generate Widget
-    return $widget->getHtmlCode(array_merge($defaultAttr, $extraAttr));
+    return $widget->getHtmlCode(array_merge(array(
+        'width' => '100%',
+        'height' => 400,
+        'frameborder' => 0
+    ), $extraAttr));
 }
 
 function fn_paymentwall_handlePingback($configs)
 {
     fn_paymentwall_initPaymentwallSdk($configs['key'], $configs['secret']);
 
+    unset($_GET['dispatch']);
     $pingback = new Paymentwall_Pingback($_GET, $_SERVER['REMOTE_ADDR']);
 
     if ($pingback->validate()) {
+
+        $paymentInfo = fn_paymentwall_getPaymentConfigs($pingback->getParameter('payment_id'));
+        $orderInfo = fn_get_order_info($pingback->getProductId());
+
         if ($pingback->getType() == PW_CREDIT_TYPE_CHARGEBACK) {
             // Update order status: Canceled
             fn_paymentwall_updateOrderStatus($pingback->getProductId(), PW_ORDER_STATUS_CANCELED);
         } else {
             // Update order status: Processed
             fn_paymentwall_updateOrderStatus($pingback->getProductId(), PW_ORDER_STATUS_PROCESSED);
+
+            // Call Delivery Confirmation API
+            if ($pingback->isDeliverable() && $paymentInfo['enable_delivery']) {
+
+                // Delivery Confirmation
+                $delivery = new Paymentwall_GenerericApiObject('delivery');
+                $response = $delivery->post(fn_paymentwall_prepare_delivery_confirmation(
+                    $orderInfo,
+                    $pingback->getParameter('ref'),
+                    $paymentInfo['test_mode']
+                ), array(
+                    "X-ApiKey: {$configs['secret']}"
```

Comment:
Remove X-ApiKey header. It set automatically in https://github.com/paymentwall/paymentwall-php/blob/master/lib/Paymentwall/ApiObject.php#L93